### PR TITLE
chore: Make parquet reader options Comet options instead of Hadoop options

### LINKED ISSUE #876: Add documentation for `COMET_PARQUET_PARALLEL_IO_ENABLED`
### What is the problem the feature request solves?

We have some undocumented configs `COMET_PARQUET_PARALLEL_IO_ENABLED` and `COMET_PARQUET_PARALLEL_IO_THREADS ` in `ReadOptions`.

They seem to be specific to the Comet native Parquet reader.

It would be nice to have some documenation. Can these be added to `CometConf`?

```
public class ReadOptions {
  private static final Logger LOG = LoggerFactory.getLogger(ReadOptions.class);
  public static final String COMET_PARQUET_PARALLEL_IO_ENABLED =
      "comet.parquet.read.parallel.io.enabled";
  public static final boolean COMET_PARQUET_PARALLEL_IO_ENABLED_DEFAULT = true;

  public static final String COMET_PARQUET_PARALLEL_IO_THREADS =
      "comet.parquet.read.parallel.io.thread-pool.size";
  public static final int COMET_PARQUET_PARALLEL_IO_THREADS_DEFAULT = 32;
```

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by parthchandra**
## Which issue does this PR close?

Closes #876 .

## Rationale for this change
Moves configuration options of the parallel reader into CometConf and adds documentation so that end users can easily understand the configuration of the reader.

## How are these changes tested?

Existing unit test

### CODE REVIEW

File: common/src/test/java/org/apache/comet/parquet/TestFileReader.java [hunk 1]
```diff
@@ -615,12 +617,12 @@ public void testColumnIndexReadWrite() throws Exception {
   @Test
   public void testWriteReadMergeScanRange() throws Throwable {
     Configuration conf = new Configuration();
-    conf.set(ReadOptions.COMET_IO_MERGE_RANGES, Boolean.toString(true));
+    conf.set(CometConf.COMET_IO_MERGE_RANGES().key(), Boolean.toString(true));
```
Comment (andygrove) [inline common/src/test/java/org/apache/comet/parquet/TestFileReader.java:620]: This test is setting the config values in a Hadoop Configuration still, and not setting them in the Spark config. Would it make sense to update the test?

File: common/src/main/java/org/apache/comet/parquet/ReadOptions.java [hunk 2]
```diff
@@ -173,14 +147,24 @@ public Builder(Configuration conf) {
       this.conf = conf;
       this.parallelIOEnabled =
           conf.getBoolean(
```
Comment (andygrove) [inline common/src/main/java/org/apache/comet/parquet/ReadOptions.java:149]: This is reading from the Hadoop conf. If I set the new configs on my Spark context, how would they get propagated to the Hadoop conf?

File: common/src/test/java/org/apache/comet/parquet/TestFileReader.java (hunk 1 above)
Comment (parthchandra) [reply common/src/test/java/org/apache/comet/parquet/TestFileReader.java:620]: There is no spark context in this test. I've added a new test with the configuration set thru the spark config

File: common/src/main/java/org/apache/comet/parquet/ReadOptions.java (hunk 2 above)
Comment (parthchandra) [reply common/src/main/java/org/apache/comet/parquet/ReadOptions.java:149]: Spark sql copies over configs that are not from spark into the hadoop config when the sql context is created. There are other settings that also use this  ( e.g. `COMET_USE_LAZY_MATERIALIZATION`, `COMET_SCAN_PREFETCH_ENABLED`)

File: spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala [hunk 3]
```diff
@@ -1280,6 +1280,81 @@ abstract class ParquetReadSuite extends CometTestBase {
...
+      expected
+    }
+
+    Seq(16, 128).foreach { batchSize =>
+      Seq(1024, 1024 * 1024).foreach { mergeRangeDelta =>
+        {
+          withSQLConf(
+            CometConf.COMET_BATCH_SIZE.key -> batchSize.toString,
+            CometConf.COMET_IO_MERGE_RANGES.key -> "true",
+            CometConf.COMET_IO_MERGE_RANGES_DELTA.key -> mergeRangeDelta.toString) {
```
Comment (andygrove) [inline spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:1335]: Is it possible to test that this config is actually making into ReadOptions? If I comment out all of the code in ReadOptions that reads these configs, the test still passes. Perhaps we just need a specific test to show that setting the config on a Spark context causes a change to the ReadOptions?

Comment (andygrove) [reply spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:1335]: I added some debug logging and I do see that it is working correctly, but would be good to have a test to confirm (and prevent regressions)

```
test is setting COMET_IO_MERGE_RANGES_DELTA = 1048576
ReadOptions ioMergeRangesDelta = 1048576
test is setting COMET_IO_MERGE_RANGES_DELTA = 1024
ReadOptions ioMergeRangesDelta = 1024
```

Comment (parthchandra) [reply spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:1336]: I added an additional check for the config. The configuration passed in to read options is not accessible but I tried to simulate the next best thing. See if that makes sense.

Comment (andygrove) [reply spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:1336]: Thanks, Parth. LGTM. Looks like you need to run `make format` to fix some import ordering.

**Pull request merged**